src/irr/asset/CGLILoader.cpp

```diff
@@ -0,0 +1,552 @@
+				case gl::INTERNAL_RGBA_ASTC_8x5: return getTranslatedFinalFormat(EF_ASTC_8x5_UNORM_BLOCK);				//GL_COMPRESSED_RGBA_ASTC_8x5_KHR
+				case gl::INTERNAL_RGBA_ASTC_8x6: return getTranslatedFinalFormat(EF_ASTC_8x6_UNORM_BLOCK);				//GL_COMPRESSED_RGBA_ASTC_8x6_KHR
+				case gl::INTERNAL_RGBA_ASTC_8x8: return getTranslatedFinalFormat(EF_ASTC_8x8_UNORM_BLOCK);				//GL_COMPRESSED_RGBA_ASTC_8x8_KHR
+				case gl::INTERNAL_RGBA_ASTC_10x5: return getTranslatedFinalFormat(EF_ASTC_10x5_UNORM_BLOCK); 				//GL_COMPRESSED_RGBA_ASTC_10x5_KHR
+				case gl::INTERNAL_RGBA_ASTC_10x6: return getTranslatedFinalFormat(EF_ASTC_10x6_UNORM_BLOCK);				//GL_COMPRESSED_RGBA_ASTC_10x6_KHR
+				case gl::INTERNAL_RGBA_ASTC_10x8: return getTranslatedFinalFormat(EF_ASTC_10x8_UNORM_BLOCK);				//GL_COMPRESSED_RGBA_ASTC_10x8_KHR
+				case gl::INTERNAL_RGBA_ASTC_10x10: return getTranslatedFinalFormat(EF_ASTC_10x10_UNORM_BLOCK);				//GL_COMPRESSED_RGBA_ASTC_10x10_KHR
+				case gl::INTERNAL_RGBA_ASTC_12x10: return getTranslatedFinalFormat(EF_ASTC_12x10_UNORM_BLOCK);				//GL_COMPRESSED_RGBA_ASTC_12x10_KHR
+				case gl::INTERNAL_RGBA_ASTC_12x12: return getTranslatedFinalFormat(EF_ASTC_12x12_UNORM_BLOCK);				//GL_COMPRESSED_RGBA_ASTC_12x12_KHR
+
+				// sRGB formats
+				case gl::INTERNAL_SRGB_DXT1: return getTranslatedFinalFormat();					//GL_COMPRESSED_SRGB_S3TC_DXT1_EXT
+				case gl::INTERNAL_SRGB_ALPHA_DXT1: return getTranslatedFinalFormat();				//GL_COMPRESSED_SRGB_ALPHA_S3TC_DXT1_EXT
+				case gl::INTERNAL_SRGB_ALPHA_DXT3: return getTranslatedFinalFormat();				//GL_COMPRESSED_SRGB_ALPHA_S3TC_DXT3_EXT
+				case gl::INTERNAL_SRGB_ALPHA_DXT5: return getTranslatedFinalFormat();				//GL_COMPRESSED_SRGB_ALPHA_S3TC_DXT5_EXT
+				case gl::INTERNAL_SRGB_BP_UNORM: return getTranslatedFinalFormat();				//GL_COMPRESSED_SRGB_ALPHA_BPTC_UNORM
+				case gl::INTERNAL_SRGB_PVRTC_2BPPV1: return getTranslatedFinalFormat();			//GL_COMPRESSED_SRGB_PVRTC_2BPPV1_EXT
+				case gl::INTERNAL_SRGB_PVRTC_4BPPV1: return getTranslatedFinalFormat();			//GL_COMPRESSED_SRGB_PVRTC_4BPPV1_EXT
+				case gl::INTERNAL_SRGB_ALPHA_PVRTC_2BPPV1: return getTranslatedFinalFormat();		//GL_COMPRESSED_SRGB_ALPHA_PVRTC_2BPPV1_EXT
+				case gl::INTERNAL_SRGB_ALPHA_PVRTC_4BPPV1: return getTranslatedFinalFormat();		//GL_COMPRESSED_SRGB_ALPHA_PVRTC_4BPPV1_EXT
+				case gl::INTERNAL_SRGB_ALPHA_PVRTC_2BPPV2: return getTranslatedFinalFormat();		//COMPRESSED_SRGB_ALPHA_PVRTC_2BPPV2_IMG
+				case gl::INTERNAL_SRGB_ALPHA_PVRTC_4BPPV2: return getTranslatedFinalFormat();		//GL_COMPRESSED_SRGB_ALPHA_PVRTC_4BPPV2_IMG
+				case gl::INTERNAL_SRGB8_ETC2: return getTranslatedFinalFormat();						//GL_COMPRESSED_SRGB8_ETC2
+				case gl::INTERNAL_SRGB8_PUNCHTHROUGH_ALPHA1_ETC2: return getTranslatedFinalFormat();	//GL_COMPRESSED_SRGB8_PUNCHTHROUGH_ALPHA1_ETC2
+				case gl::INTERNAL_SRGB8_ALPHA8_ETC2_EAC: return getTranslatedFinalFormat();			//GL_COMPRESSED_SRGB8_ALPHA8_ETC2_EAC
```

Comment:
all here https://github.com/buildaworldnet/IrrlichtBAW/blob/master/include/irr/asset/format/EFormat.h#L196

PUNCHTHROUGH means 1-bit alpha (on or off, like punching through a card)

Reply:
okay, got it

I don't understand, I have to assign `EF_ETC2_R8G8B8_UNORM_BLOCK` to `INTERNAL_SRGB8_ALPHA8_ETC2_EAC`, but as I can imagine `EF_ETC2_R8G8B8_UNORM_BLOCK` doesn't have any alpha channel, hence why do I assign it to it?

Reply:
got it